index.html

```diff
@@ -126,187 +846,604 @@ <h3>
-            {{SocketOptions/localAddress}} and {{SocketOptions/localPort}}.
-          </div>
         </section>
-        <section data-dfn-for="TCPSocketOptions">
-          <h3>
-            `TCPSocketOptions` dictionary
-          </h3>
+
+      </section>
+
+      <section>
+        <h3>{{UDPSocket/[[readable]]}} attribute (internal)</h3>
+
+        <aside class="example">
+          An application receiving data from a UDP socket will typically use the {{ReadableStream}} like this:
+
+          <pre class="js">
+            const { readable } = await socket.connection;
+
+            const reader = readable.getReader();
+
+            while (true) {
+              const { value, done } = await reader.read();
+              if (done) {
+                // |reader| has been canceled.
+                break;
+              }
+              const { data } = value;
+              // Do something with |data|...
+            }
+
+            reader.releaseLock();
+          </pre>
+        </aside>
+
+        The steps to <dfn data-lt="initialize udp readable stream">initialize the UDP readable stream</dfn> are:
+
+        <ol>
+          <li>Let |stream:ReadableStream| be a [=new=] {{ReadableStream}}.
+          <li>
+            Let |pullAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Let |desiredSize| be the
+                <a href="https://streams.spec.whatwg.org/#desired-size-to-fill-a-streams-internal-queue">desired size</a>
+                of [=this=].{{UDPSocket/[[readable]]}}'s
+                <a href="https://streams.spec.whatwg.org/#internal-queues">internal queue</a>.
+              <li>
+                Run the following steps |desiredSize| times [=in parallel=].
+                <ol>
+                  <li>
+                    Invoke the operating system to read a UDP packet from the socket, placing the result in the [=byte sequence=] |bytes|.
+                  <li>
+                    [=Queue a global task=] on the [=relevant global object=] of
+                    [=this=] using the [=UDP socket task source=] to run the
+                    following steps:
+                    <ol>
+                      <li>
+                        If no errors were encountered run the following steps:
+                        <ol>
+                          <li>
+                            Let |buffer| be a [=new=] {{ArrayBuffer}} created from |bytes|.
+                          <li>
+                            Let |chunk| be a [=new=] {{Uint8Array}} view over |buffer|,
+                            who's length is the length of |bytes|.
+                          <li> 
+                            Let |message:UDPMessage| be a new |UDPMessage|.
+                          <li>
+                            Set |message|.{{UDPMessage/data}} to |chunk|.
+                          <li>
+                            Infer the origins of the received UDP packet and populate
```

Comment:
I would rather ignore these fields completely for now.

Reply:
I think I like "remote" better because in both the sending and receiving case that's the part which you care about. You know what the "local" address and port are because that's what you bound to.